### PR TITLE
[CI] Skip dev-igc run if the PR has `ci-no-devigc` label specified

### PR DESCRIPTION
**Pull request by uditagarwal97**
Currently, if `ci-no-devigc` label is specified, we duplicate the E2E test run with `ghcr.io/intel/llvm/ubuntu2204_intel_drivers:latest` image.
This PR intends to skip the run instead of duplicating.

### CODE REVIEW

**Review by aelovikov-intel — APPROVED** [review]

**Comment by sarnex** [comment]
@uditagarwal97 LGTM, can you just post the log from a run of dev-igc before the ci-dev-igc label was set, I want to make sure that case is working but I can't find the logs :(

**Comment by uditagarwal97** [comment]
> @uditagarwal97 LGTM, can you just post the log from a run of dev-igc before the ci-dev-igc label was set, I want to make sure that case is working but I can't find the logs :(

That's weird. Even I can't find the logs from before. I can remove this label and re-open the PR again, if that helps.

File: .github/workflows/sycl-linux-precommit.yml [hunk 1]
```diff
@@ -110,6 +110,8 @@ jobs:
             use_dev_igc: ${{ contains(needs.detect_changes.outputs.filters, 'devigccfg') }}
             extra_lit_opts: --param matrix-xmx8=True --param gpu-intel-dg2=True
             env: '{"LIT_FILTER":${{ needs.determine_arc_tests.outputs.arc_tests }} }'
+            # Run only if the PR do not have the 'ci-no-devigc' label.
```
Comment (sarnex) [inline .github/workflows/sycl-linux-precommit.yml:113]:
```suggestion
            # Run only if the PR does not have the 'ci-no-devigc' label.
```

**Review by sarnex — APPROVED** [review]
seems working as expected, LGTM minus a typo :)

File: .github/workflows/sycl-linux-precommit.yml [hunk 2]
```diff
@@ -110,6 +110,8 @@ jobs:
             use_dev_igc: ${{ contains(needs.detect_changes.outputs.filters, 'devigccfg') }}
             extra_lit_opts: --param matrix-xmx8=True --param gpu-intel-dg2=True
             env: '{"LIT_FILTER":${{ needs.determine_arc_tests.outputs.arc_tests }} }'
+            # Run only if the PR does not have the 'ci-no-devigc' label.
+            run_condition: ${{ contains(github.event.pull_request.labels.*.name, 'ci-no-devigc') && 'false' || 'true' }}
```
Comment (jsji) [inline .github/workflows/sycl-linux-precommit.yml:114]: How about "skip-run:  ${{ contains(github.event.pull_request.labels.*.name, 'ci-no-devigc')"

File: .github/workflows/sycl-linux-precommit.yml [hunk 3]
```diff
@@ -123,6 +125,7 @@ jobs:
       use_dev_igc: ${{ matrix.use_dev_igc }}
       extra_lit_opts: ${{ matrix.extra_lit_opts }}
       env: ${{ matrix.env || '{}' }}
+      run_condition: ${{ matrix.run_condition || 'true' }}
```
Comment (jsji) [inline .github/workflows/sycl-linux-precommit.yml:128]: skip_run: ${{ matrix.skip_run }}

File: .github/workflows/sycl-linux-run-tests.yml [hunk 4]
```diff
@@ -76,6 +76,11 @@ on:
         default: '{}'
         required: False
 
+      run_condition:
```
Comment (jsji) [inline .github/workflows/sycl-linux-run-tests.yml:79]: "skip-run:"

File: .github/workflows/sycl-linux-run-tests.yml [hunk 5]
```diff
@@ -146,6 +151,7 @@ permissions:
 
 jobs:
   run:
+    if: inputs.run_condition == 'true'
```
Comment (jsji) [inline .github/workflows/sycl-linux-run-tests.yml:154]: if: inputs.run_condition == 'false'

File: .github/workflows/sycl-linux-run-tests.yml [hunk 6]
```diff
@@ -76,6 +76,11 @@ on:
         default: '{}'
         required: False
 
+      run_condition:
+        type: string
+        default: 'true'
```
Comment (jsji) [inline .github/workflows/sycl-linux-run-tests.yml:81]: default:  'false'

File: .github/workflows/sycl-linux-precommit.yml (hunk 2 above)
Comment (uditagarwal97) [reply .github/workflows/sycl-linux-precommit.yml]: Fixed in 01c5a4df09765afb0069410e2feec8fe257effc6

File: .github/workflows/sycl-linux-run-tests.yml (hunk 4 above)
Comment (uditagarwal97) [reply .github/workflows/sycl-linux-run-tests.yml]: Fixed in 01c5a4df09765afb0069410e2feec8fe257effc6

File: .github/workflows/sycl-linux-precommit.yml [hunk 7]
```diff
@@ -110,6 +110,8 @@ jobs:
             use_dev_igc: ${{ contains(needs.detect_changes.outputs.filters, 'devigccfg') }}
             extra_lit_opts: --param matrix-xmx8=True --param gpu-intel-dg2=True
             env: '{"LIT_FILTER":${{ needs.determine_arc_tests.outputs.arc_tests }} }'
+            # Run only if the PR does not have the 'ci-no-devigc' label.
+            skip_run: ${{ contains(github.event.pull_request.labels.*.name, 'ci-no-devigc') && 'true' || 'false' }}
```
Comment (jsji) [inline .github/workflows/sycl-linux-precommit.yml:114]: Do we still need the ( && 'true' || 'false')?

**Review by jsji — APPROVED** [review]
LGTM. Thanks.

**Review by YuriPlyakhin — APPROVED** [review]

File: .github/workflows/sycl-linux-precommit.yml (hunk 7 above)
Comment (uditagarwal97) [reply .github/workflows/sycl-linux-precommit.yml:114]: Yes, because `skip_run` is of type String

Comment (uditagarwal97) [reply .github/workflows/sycl-linux-precommit.yml:114]: Actually, you are right. Now that I refer to GHA docs (https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#contains), `contains()` return `true`/`false` strings. Previously, I though `contains()` returns boolean values.

**Comment by uditagarwal97** [comment]
Test failure on Arc is unrelated. I've opened a issue for it: https://github.com/intel/llvm/issues/15321
@intel/llvm-gatekeepers The PR is ready.

**Pull request merged**